### ISSUE TITLE
Airsonic 10.6.0 Release Plan

### ISSUE THREAD

**Issue by fxthomas** [airsonic/airsonic#1438]
This issue is a meta issue for discussing the Airsonic 10.6.0 release. I'll keep this up-to-date as we add more information to the discussion.

This won't change how small PRs and issues are merged, but with this I hope we can be a little more open about how the releases are done. These meta issues were discussed in #1436.

Things to do:

- [x] Important features / fixes (https://github.com/airsonic/airsonic/milestone/6)
  - [x] Bring back the ability to seek in streams (#1123) 
  - [x] Add a minimal "system check" page (#1082)
  - [x] Handle issues with ordering of index/scans at startup (#1444 and #1402)
  - [ ] ~~Provide a first workaround for password security (#69)~~
- [x] Prepare release
  - [x] Follow the [release documentation](https://github.com/airsonic/airsonic/blob/master/contrib/release.md)
- [x] Publish release
  - [x] Publish release on GitHub
  - [x] Push release on Docker Hub
  - [x] Write post on Reddit
- [x] Post-release
  - [x] Write meta issue for next release

**Comment by fxthomas** [comment]
Ideally I'd like to have at least #1123 in 10.6, this will bring Airsonic to a point where it is quite stable before we do big, potentially breaking changes in 11.0.

**Comment by eharris** [comment]
I'd also like to get #1124 and some additional transcoding cleanups in if possible.  I haven't really done much additional work past the already open PR's because there are a lot of interdependencies and didn't really seem worth the effort (or dealing with the inevitable merge conflicts) until some of the initial work got merged.

**Comment by tesshucom** [comment]
I welcome you rather than oppose what is stated in the release plan.
I also know that they have a very long review period.
Nevertheless, I don't think it's preferable to overwrite the current master.

The current master seems to be neglecte stability.
Numerous degrated bugs have been reported.

Recent search and scan bug reports are fundamentally related to the following PRs:

#1376, #1262

These are not simple enhancements.
It includes specification changes and implementation changes, but has not been adequately reviewed and tested.

___
muff1nman said
> just because someone thinks an issue should get addressed in a release, doesn't mean that someone is going to do it. That partly explains why the current milestone tracking is very light.

Surely.
In parallel with the addition of new features, you can proceed while solving some existing small problems.

However, new bugs or functionality degradation due to changes should not be treated as equivalent to known bugs.
In a project that overlooks them, it is impossible to perform defect correction and function addition in parallel.

___

I know that db and Spring updates are meaningful.
But even if new problems related to the core function are ignored and db and Spring are performed, the benefit to the user is limited.

It is recommended that you make it a mandatory item in 10.6.0 or create a patch version as soon as possible.
If repair is difficult, we should revert and check the operation to eliminate the problem.

**Comment by fxthomas** [comment]
If I read it correctly there is a chance that #1124 breaks some setups (people who modified the downsampling command), and in that regard it is a breaking change. Since 11.0 is close anyway, I would rather keep it for just after 10.6 if that's okay with you.

@tesshucom I'm okay with fixing #1444 in 10.6.0, I think it isn't good if the first thing somebody sees in Airsonic is a big crash in the logs. Can you add a note on the issue if you start working on this? I'll do it myself if nobody does it before the release.

I'll move these two on 11.0 and 10.6 milestones respectively for now.

**Comment by tesshucom** [comment]
Added a little to #1444.

This is not a coercion or order, but a idea.

If things such as what to do, what not to do, when to respond and Airsonic's policies are a little clearer, I think that it will be possible to subdivide and visualize tasks and get collaborators.

**Comment by fxthomas** [comment]
Also, it's been on my mind for a while, how about adding a first iteration of https://github.com/airsonic/airsonic/issues/1082?

It's been bugging me forever that people don't have a clean way of providing detailed information about their system in their bug reports, and this should be fairly easy to implement.

**Comment by tesshucom** [comment]
Is #1340 not important?

Nobody will want corporate quality for Airsonic.
However, speed degradation are a fatal bug in general development.
Usually, these are investigated and the countermeasures taken.
Or may be rolled back.

The expectation that this will be corrected in future changes is not convincing.
Because you shouldn't merge instability fixes in the first place.
Or by any chance, if you notice degradation, you need to take immediate action.
Alternatively, there is a method to announce that the external DB is a WIP function unlike Subsonic.

If your project prioritizes the development of new features over stability, say no more.

Previously randomnicode had a constructive proposal.
Rather than waiting for others to follow up on the master branch, why not create a branch every night or do a pre-release for months?
The quality of 10.4.1-10.5.0 is close to such artifacts.

**Comment by eharris** [comment]
I think #1131 should also be included, as it resolves a longstanding security issue.

**Comment by fxthomas** [comment]
I did not follow it too closely, are there any breaking changes? If not then we could add it.

**Comment by fxthomas** [comment]
Additionally, I initially wanted to add something to prepare for the changes in #69, but it looks like I won't have time to finish it properly soon, so it'll be postponed to 11.0.

**Comment by tesshucom** [comment]
> Handle issues with ordering of index/scans at startup (#1444 and #1402)

I think #1550 is necessary because of these issues and data integrity issues.

#1557 and #1555 are health check patches for new features, so simultaneous release is desirable.

These are travis errors now, because the original commit is a branch point for tracing the history.
If we don't care about history, rebasing is fine!

**Comment by jvoisin** [comment]
I would vote for not releasing with red tests, but OTOH, I don't have much time to allocate to airsonic this month :/

**Comment by fxthomas** [comment]
I'm going to try my hand at using the proposed solution and making it work for us, it's not like I can easily move out of home right now anyway ;)

**Comment by eharris** [comment]
I didn't see an obvious response to this, so to remind, can we include #1131?

**Comment by tesshucom** [comment]
#1131 appears to have raised doubts.
The review does not appear to be complete.

In #1539, randomnicode seems to have made an interesting point in the post-merge verification.
(If it is not fatal, another PR may be fine)

> I would vote for not releasing with red tests

I agree rather.
However realistically, it seems that there is no big problem even if Java11 is temporarily excluded from travis verification in case of prolongation.
(Obvious runtime error, so validating dependencies is the right path.)

#1240 seems to be the story after that.

**Comment by fxthomas** [comment]
I am getting way more time than I initially expected right now due to the confinement, so I'll help with #1131 as well. I don't see any issue with it as it is, so let me give it a whirl on my instance for a few days. If everything works fine I see no issue with merging it.

@tesshucom Good point re: the play queue thing. I created an issue with that comment (it's much easier to keep track of than a small comment in a merged PR). It's been this way for as long as I remember however, so not fixing it immediately is fine.

**Comment by fxthomas** [comment]
Okay, so I was able to find out how to fix the Docker issue in #1604, plus a fix for a new CVE in #1603. Still not sure why that #1604 was necessary, how dependencies are managed by Maven is still a bit of a mystery to me.

@eharris I added #1131 to the milestone, since I saw no issues with it I think we can safely add it.

**Comment by tesshucom** [comment]
> I don't think the negligible security gained here is worth losing the feature over.

I am on the same side as muff1nman.
I think this is not a closed room agreement, but a type of functional change that requires careful explanation to the outside.

Isn't this a type of feature change that can't be released without documentation updates?
As a general feeling.

**Comment by tesshucom** [comment]
There is no problem putting #1131 on a milestone.

> I don't think the negligible security gained here is worth losing the feature over.

I am on the same side as muff1nman.
I think this is not a closed room agreement, but a type of functional change that requires careful explanation to the outside.
Isn't this a type of feature change that can't be released without documentation updates?
As a general feeling.

**Comment by fxthomas** [comment]
>  Isn't this a type of feature change that can't be released without documentation updates?

I looked at it the other day, and muff1nman's comment was about a previous issue with the PR that was fixed later.

In its current state, if the user does not remove the default admin user nothing changes. And I saw no issue with removing the `admin` user on my instance, even when importing playlists, so it seems to be working as intended without breaking changes. For me it's good to go! :wink:

**Comment by fxthomas** [comment]
Everything is merged and passing, I'll prepare the release in the next days!

**Comment by tesshucom** [comment]
#1565 seems that the specification has been changed ?

There is no objection to disabling legacy features, but so far it is customary to have a set of appropriate HTTP errors and documentation.

[API:NOTE](https://airsonic.github.io/docs/api/)

I don't know who maintains the API, but if you're not sure, ask a muff1nman.

**Comment by fxthomas** [comment]
Apparently Chartlyrics being down was just a random issue on their part? I don't like losing a feature if we can avoid it, @eharris how about we revert the changes in the PR?

**Comment by fxthomas** [comment]
I wrote a draft release notes here: https://github.com/airsonic/airsonic/releases/tag/untagged-96adb9574179686be54c (in the Releases page), let me know what you think.

**Comment by Faeb35** [comment]
Thank you guys! I'm so excited :-)

**Comment by tesshucom** [comment]
> I wrote a draft release notes 

I think there is no problem.

 - Settle #1565
 - Jquery alert is occurring in HEAD. https://blog.jquery.com/2020/04/10/jquery-3-5-0-released/
   Update to 3.5 or suppress. I believe that there will be no impact after updating, but it is safe if experts follow us.

**Comment by fxthomas** [comment]
I don't believe 3.5 should be an issue given what they show on the changelog. Let me run that tomorrow and I'll merge it if I don't see any issue.

**Comment by tesshucom** [comment]
I didn't look very closely.

https://github.com/airsonic/airsonic/pull/1585

It seems that the order of the algorithms is not taken into account.
Depending on the specifications of the VM, it may or may not work as expected.

**Comment by tesshucom** [comment]
I didn't look very closely.

https://github.com/airsonic/airsonic/pull/1585

It seems that the order of the algorithms is not taken into account.
Depending on the specifications of the VM, it may or may not work as expected.
(The processing flow is more important.)

**Comment by fxthomas** [comment]
Do you have a concrete example of something that does not work? Can you submit a PR?

**Comment by tesshucom** [comment]
I submitted a PR.
If you always want to use SHA1PRNG, no modification is necessary ...

**Comment by fxthomas** [comment]
I chose to merge #1614 so that we can keep the existing functionality. Will do the release tonight if nothing else stands in the way!

**Comment by fxthomas** [comment]
Everything is done, let's rock on to the next release!
